### PR TITLE
statistics: set the right range for tidb_analyze_partition_concurrency

### PR DESCRIPTION
**Pull request by Rustin170506**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

We have defined the correct range for tidb_analyze_partition_concurrency in [this pull request](https://github.com/pingcap/tidb/pull/55409).

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [ ] v8.3 (TiDB 8.3 versions)
- [ ] v8.2 (TiDB 8.2 versions)
- [ ] v8.1 (TiDB 8.1 versions)
- [ ] v8.0 (TiDB 8.0 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by songrijie — APPROVED** [review]

**Comment by Oreoxmt** [comment]
@Rustin170506 Could you please help confirm which version this change applies to - v8.3 or v8.4?

**Comment by Rustin170506** [comment]
> @Rustin170506 Could you please help confirm which version this change applies to - v8.3 or v8.4?

v8.4

File: system-variables.md [hunk 1]
```diff
@@ -1128,7 +1128,8 @@ MPP is a distributed computing framework provided by the TiFlash engine, which a
 - Persists to cluster: Yes
 - Applies to hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value): No
 - Default value: `2`. The default value is `1` for v7.4.0 and earlier versions.
-- This variable specifies the concurrency of reading and writing statistics for a partitioned table when TiDB analyzes the partitioned table.
+- Range: `[1, 128]`
+- This variable specifies the concurrency for writing collected statistics for a partitioned table when TiDB analyzes it.
```
Comment (Oreoxmt) [inline system-variables.md:1132]:
```suggestion
- This variable specifies the concurrency for writing collected statistics when TiDB analyzes a partitioned table.
```

File: system-variables.md [hunk 2]
```diff
@@ -1128,7 +1128,8 @@ MPP is a distributed computing framework provided by the TiFlash engine, which a
 - Persists to cluster: Yes
 - Applies to hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value): No
 - Default value: `2`. The default value is `1` for v7.4.0 and earlier versions.
-- This variable specifies the concurrency of reading and writing statistics for a partitioned table when TiDB analyzes the partitioned table.
+- Range: `[1, 128]`
```
Comment (Oreoxmt) [inline system-variables.md:1131]:
```suggestion
- Range: `[1, 128]`. Before v8.4.0, the value range is `[1, ]`.
```
Please provide the value range for v8.3.0 and earlier versions

Comment (Oreoxmt) [reply system-variables.md:1131]: @Rustin170506 PTAL

File: system-variables.md [hunk 3]
```diff
@@ -1128,7 +1128,8 @@ MPP is a distributed computing framework provided by the TiFlash engine, which a
 - Persists to cluster: Yes
 - Applies to hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value): No
 - Default value: `2`. The default value is `1` for v7.4.0 and earlier versions.
-- This variable specifies the concurrency of reading and writing statistics for a partitioned table when TiDB analyzes the partitioned table.
+- Range: `[1, 128]`. Before v8.4.0, the value range is `[1, ]`.
```
Comment (Oreoxmt) [inline system-variables.md:1131]: What is the maximum value for v8.3.0 and earlier versions?

Comment (Oreoxmt) [reply system-variables.md:1131]: @Rustin170506 PTAL

**Review by Rustin170506 — COMMENTED** [review]
🔢 Self-check (PR reviewed by myself and ready for feedback.)

**Review by Oreoxmt — APPROVED** [review]

**Review by qiancai — APPROVED** [review]

**Comment by Rustin170506** [comment]
@Oreoxmt Can we merge it now? Thanks!

**Pull request merged**